rust/agama-server/src/web/service.rs

```diff
@@ -85,8 +88,64 @@ impl MainServiceBuilder {
             .route("/auth", post(login).get(session).delete(logout));
 
         let serve = ServeDir::new(self.public_dir);
+
+        // handle the /po.js request
+        // the requested language (locale) is sent in the "agamaLanguage" HTTP cookie
+        // this reimplements the Cockpit translation support
+        async fn po(jar: CookieJar) -> impl IntoResponse {
+            let mut response_headers = HeaderMap::new();
+
+            if let Some(cookie) = jar.get("agamaLanguage") {
+                let mut target_file = String::new();
+                let mut found = false;
+                // FIXME: this does not work, the public_dir setting is not accessible :-/
+                // when using something like PathBuf::from("/usr/share/cockpit/agama") here
+                // it works just fine....
+                let prefix = self.public_dir;
```

Comment:
Here I need to somehow access the `self.public_dir` setting. I have no idea how to do that. Use a closure? Implement an axum service?

Any idea Rust experts?